pkg/controllers/repositoryserver/utils.go

```diff
@@ -0,0 +1,269 @@
+	repoServerPod     = "repo-server-pod"
+
+	repoServerServiceNameKey  = "name"
+	repoServerServiceProtocol = "TCP"
+	repoServerServicePort     = 51515
+	repoServerAddressFormat   = "https://%s:%d"
+	repoServerUsernameFormat  = "%s@%s"
+
+	repoServerPodContainerName    = "repo-server-container"
+	googleCloudCredsDirPath       = "/mnt/secrets/creds/gcloud"
+	googleCloudServiceAccFileName = "service-account.json"
+
+	// CustomCACertName is the name of the custom root CA certificate
+	customCACertName        = "custom-ca-bundle.pem"
+	tlsCertVolumeName       = "kopia-cert"
+	tlsCertDefaultMountPath = "/mnt/secrets/tlscert"
+	tlsKeyPath              = "/mnt/secrets/tlscert/tls.key"
+	tlsCertPath             = "/mnt/secrets/tlscert/tls.crt"
+)
+
+func getRepoServerService(namespace string) corev1.Service {
+	name := fmt.Sprintf("%s-%s", repoServerService, rand.String(5))
+	return corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    map[string]string{repoServerServiceNameKey: name},
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Name:     fmt.Sprintf("%s-port", repoServerService),
+					Protocol: repoServerServiceProtocol,
+					Port:     repoServerServicePort,
+				},
+			},
+			Selector: map[string]string{repoServerServiceNameKey: name},
+		},
+	}
+}
+
+// RepoServerNetworkPolicy returns a network policy with appropriate pod selector
+func getRepoServerNetworkPolicy(
+	namespace string,
+	svc *corev1.Service,
+	podSelector *metav1.LabelSelector,
+	namespaceSelector *metav1.LabelSelector,
+) *networkingv1.NetworkPolicy {
+	protocolTCP := corev1.ProtocolTCP
+	port := intstr.FromInt(repoServerServicePort)
+	return &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: fmt.Sprintf("%s-", repoServerNP),
+			Namespace:    namespace,
+			Labels:       map[string]string{repoServerServiceNameKey: svc.Name},
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{repoServerServiceNameKey: svc.Name},
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{
+				{
+					From: []networkingv1.NetworkPolicyPeer{
+						{
+							PodSelector:       podSelector,
+							NamespaceSelector: namespaceSelector,
+						},
+					},
+					Ports: []networkingv1.NetworkPolicyPort{
+						{
+							Protocol: &protocolTCP,
+							Port:     &port,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getPodOverride(ctx context.Context, reconciler *RepositoryServerReconciler, namespace string) (map[string]interface{}, error) {
+	podName := os.Getenv("HOSTNAME")
+	pod := corev1.Pod{}
+	err := reconciler.Get(ctx, types.NamespacedName{Name: podName, Namespace: namespace}, &pod)
+	if err != nil {
+		return nil, err
+	}
+	uidguid := int64(0)
+	nonRootBool := false
+	podOverride := map[string]interface{}{
+		"securityContext": corev1.PodSecurityContext{
```

Comment:
Since now `PodSecurityContext` is a part of PodOptions, we can set it there as well. This can be done in a follow-up as well.